dpdispatcher/dpcloudserver/api.py

```diff
@@ -118,6 +119,47 @@ def job_create(job_type, oss_path, input_data, program_id=None):
     return ret['job_id']
 
 
+def job_create_v2(job_type, oss_path, input_data, program_id=None, group_id=None):
+    post_data = {
+                'job_type': job_type,
+                'oss_path': oss_path,
+            }
+    if program_id is not None:
+        post_data["program_id"] = program_id
+    if input_data.get('job_group_id') is not None:
+        post_data["job_group_id"] = input_data.get('job_group_id')
+    if group_id is not None:
+            post_data["job_group_id"] = group_id
+    if input_data.get('job_name') is not None:
+        post_data["job_name"] = input_data.get('job_name')
+    if input_data.get('rerun') is not None:
+        post_data["rerun"] = input_data.get('rerun')
+    if input_data.get('image_name') is not None:
+        post_data["image_name"] = input_data.get('image_name')
+    if input_data.get('disk_size') is not None:
+        post_data["disk_size"] = input_data.get('disk_size')
+    if input_data.get('scass_type') is not None:
+        post_data["scass_type"] = input_data.get('scass_type')
+    if input_data.get('instance_group_id') is not None:
+        post_data["instance_group_id"] = input_data.get('instance_group_id')
+    if input_data.get('command') is not None:
+        post_data["cmd"] = input_data.get('command')
+    if input_data.get('log_file') is not None:
+        post_data["log_file"] = input_data.get('log_file')
+    if input_data.get('backward_files') is not None:
+        post_data["out_files"] = input_data.get('backward_files')
+    if input_data.get('platform') is not None:
+        post_data["platform"] = input_data.get('platform')
+    if input_data.get('region') is not None:
+        post_data["region"] = input_data.get('region')
+    if input_data.get('zone') is not None:
+        post_data["zone"] = input_data.get('zone')
+    if input_data.get('on_demand') is not None:
+        post_data["on_demand"] = input_data.get('on_demand')
```

Comment:
I suggest to write as
```py
for key in ('job_group_id', 'job_name', 'rerun', 'image_name',): #...
    if input_data.get(key) is not None:
        post_data[key] = input_data.get(key)
```

Reply:
thank you for the suggestion, I rewrote it in the latest commit.